sklearn/impute/_base.py

```diff
@@ -52,9 +52,7 @@ def _most_frequent(array, extra_value, n_repeat):
                 if count == most_frequent_count
             )
         else:
-            mode = stats.mode(array)
-            most_frequent_value = mode[0][0]
-            most_frequent_count = mode[1][0]
+            most_frequent_value, most_frequent_count = stats.mode(array)
```

Comment:
To be specific, I think we need to work around SciPy's deprecation warning by placing this into `utils.fixes` and adjust the code that uses mode.

```python
def _mode(a, axis=0):
    if sp_version >= parse_version("1.9.0"):
        return stats.mode(a, keepdims=False)
    
    # unpack for SciPy version < 1.9
    results = stats.mode(a)
    return results[0][0], results[1][0]
```

When our min supported SciPy version is 1.9 we can call `stats.mode` directly with `keepdims=False`.

Reply:
I opened a PR to your fork with this idea: https://github.com/glemaitre/scikit-learn/pull/12